iohk-monitoring/src/Cardano/BM/Output/Aggregation.lhs

```diff
@@ -263,50 +293,62 @@ We use Welford's online algorithm to update the estimation of mean and variance
 (see \url{https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Welford's_Online_algorithm})
 
 \begin{code}
-updateAggregation :: Measurable -> Aggregated -> LOMeta -> Maybe Int -> Aggregated
+updateAggregation :: Measurable -> Aggregated -> LOMeta -> Maybe Word64 -> Either Text Aggregated
 updateAggregation v (AggregatedStats s) lme resetAfter =
     let count = fcount (fbasic s)
         reset = maybe False (count >=) resetAfter
     in
     if reset
     then
-        singletonStats v
+        Right $ singletonStats v
     else
-        AggregatedStats $! Stats { flast  = v
-                                 , fold = mkTimestamp
-                                 , fbasic = updateBaseStats (count >= 1) v (fbasic s)
-                                 , fdelta = updateBaseStats (count >= 2) (v - flast s) (fdelta s)
-                                 , ftimed = updateBaseStats (count >= 2) (mkTimestamp - fold s) (ftimed s)
-                                 }
+        Right $ AggregatedStats $! Stats { flast  = v
+                                         , fold = mkTimestamp
+                                         , fbasic = updateBaseStats 1 v (fbasic s)
+                                         , fdelta = updateBaseStats 2 deltav (fdelta s)
+                                         , ftimed = updateBaseStats 2 timediff (ftimed s)
+                                         }
   where
+    deltav = subtractMeasurable v (flast s)
     mkTimestamp = utc2ns (tstamp lme)
+    timediff = Nanoseconds $ fromInteger $ (getInteger mkTimestamp) - (getInteger $ fold s)
     utc2ns (UTCTime days secs) =
```

Comment:
`utc2ns` is also defined here:
https://github.com/input-output-hk/iohk-monitoring-framework/pull/303/files#diff-cbc1ae38e6be73f27a17b5791e7cb91aR180
Should we put it into another file in order to both use it from there?

Reply:
yes, this could be done if we find a good place for it.